src/main/java/io/confluent/connect/hdfs/wal/FSWAL.java

```diff
@@ -99,6 +99,7 @@ public void acquireLease() throws ConnectException {
           throw new ConnectException(e);
         }
       } catch (IOException e) {
+        close();
```

Comment:
do we have to call close in the RemoteExceptionBlock also specially when we are throwing ConnectException?

Reply:
Also what about below block where you acquire the writer but we are throwing    

```
 if (sleepIntervalMs >= WALConstants.MAX_SLEEP_INTERVAL_MS) {
      throw new ConnectException("Cannot acquire lease after timeout, will retry.");
}
```

Reply:
+1 that `close()` may need to be called in the `RemoteExceptionBlock` before throwing a `ConnectException`.

Reply:
One additional thought about calling `close()` here (that may not be an issue) is that `close()` also closes the reader if it's non-null, but in this `acquireLease()` method we only call methods on the writer.

Reply:
Good catch @kpatelatwork! You are right, whenever we call
```
writer = WALFile.createWriter(conf, Writer.file(new Path(logFile)),
                                        Writer.appendIfExists(true));
```
it creates a cache entry, and we should always close it if applicable in case we are throwing an exception.
I moved `close()` to a `finally` block.

@dosvath right, `acquireLease()` only creates `writer`. Do you think it makes sense to split `close()` into `closeReader()` and `closeWriter()` and call `closeWriter()` from `acquireLease()`?